src/Altinn.Notifications.Integrations/Register/RegisterClient.cs

```diff
@@ -1,52 +1,105 @@
+
+        if (!response.IsSuccessStatusCode)
+        {
+            throw await PlatformHttpException.CreateAsync(response);
+        }
+
+        var responseContent = await response.Content.ReadAsStringAsync();
+        var contactPoints = JsonSerializer.Deserialize<OrgContactPointsList>(responseContent, _jsonSerializerOptions)?.ContactPointsList;
+        return contactPoints ?? [];
+    }
 
-            HttpContent content = new StringContent(JsonSerializer.Serialize(lookupObject, JsonSerializerOptionsProvider.Options), Encoding.UTF8, "application/json");
+    /// <inheritdoc/>
+    public async Task<List<PartyDetails>> GetPartyDetails(List<string> organizationNumbers, List<string> socialSecurityNumbers)
+    {
+        if ((organizationNumbers?.Count ?? 0) == 0 && (socialSecurityNumbers?.Count ?? 0) == 0)
+        {
+            return [];
+        }
 
-            var response = await _client.PostAsync("organizations/contactpoint/lookup", content);
+        var partyDetailsLookupBatch = new PartyDetailsLookupBatch(organizationNumbers, socialSecurityNumbers);
+        var content = CreateJsonContent(partyDetailsLookupBatch);
 
-            if (!response.IsSuccessStatusCode)
-            {
-                throw await PlatformHttpException.CreateAsync(response);
-            }
+        var response = await _client.PostAsync(_nameComponentsLookupEndpoint, content);
 
-            string responseContent = await response.Content.ReadAsStringAsync();
-            List<OrganizationContactPoints>? contactPoints = JsonSerializer.Deserialize<OrgContactPointsList>(responseContent, JsonSerializerOptionsProvider.Options)!.ContactPointsList;
-            return contactPoints!;
+        if (!response.IsSuccessStatusCode)
+        {
+            throw await PlatformHttpException.CreateAsync(response);
         }
+
+        var responseContent = await response.Content.ReadAsStringAsync();
+        var partyDetailsList = JsonSerializer.Deserialize<PartyDetailsLookupResult>(responseContent, _jsonSerializerOptions)?.PartyDetailsList;
+
+        return partyDetailsList ?? [];
+    }
+
+    /// <summary>
+    /// Creates an HTTP content object with a JSON-serialized representation of the specified object.
+    /// </summary>
+    /// <typeparam name="T">The type of the object to serialize.</typeparam>
+    /// <param name="payload">The object to serialize into JSON.</param>
+    /// <returns>
+    /// An <see cref="HttpContent"/> object containing the serialized JSON representation 
+    /// of the provided object, encoded in UTF-8, with a media type of "application/json".
+    /// </returns>
+    /// <remarks>
+    /// This method uses the specified <see cref="JsonSerializerOptions"/> to control the serialization behavior.
+    /// </remarks>
+    private StringContent CreateJsonContent<T>(T payload)
+    {
+        var json = JsonSerializer.Serialize(payload, _jsonSerializerOptions);
+        return new StringContent(json, Encoding.UTF8, "application/json");
```

Comment:
This is how we do it in a lot of cases, but for future reference, this can be done for us by existing methods. There is a class called `JsonContent` in the `System.Net.Http.Json` namespace.

Reply:
No need to make any changes now, but consider this for future features and rewrites.